src/javascripts/ng-admin/Crud/field/maReferenceField.js

```diff
@@ -0,0 +1,87 @@
+function maReferenceField($compile, ReadQueries) {
+    return {
+        scope: {
+            'field': '&',
+            'value': '=',
+            'entry':  '=?',
+            'datastore': '&?'
+        },
+        restrict: 'E',
+        compile: function() {
+            return {
+                pre: function(scope, element) {
+                    var field = scope.field();
+                    scope.name = field.name();
+                    scope.v = field.validation();
+
+                    scope.choices = [];
+
+                    var valueFieldName = field.targetEntity().identifier().name()
+                    var labelFieldName = field.targetField().name();
+
+                    scope.refreshChoices = function(search) {
+                        var referenceFields = {};
+                        referenceFields[scope.name] = field;
+
+                        return ReadQueries.getAllReferencedData(referenceFields, search)
+                            .then(r => r[field.name()])
+                            .then(results => {
+                                return results.map(function(r) {
+                                    return {
+                                        value: r[valueFieldName],
+                                        label: field.getMappedValue(r[labelFieldName], r)
+                                    };
+                                });
+                            })
+                            .then(formattedResults => {
+                                scope.choices = formattedResults;
+                                scope.$root.$$phase || scope.$apply();
+                            });
+                    };
+
+                    scope.refreshDelay = field.refreshDelay();
+
+                    var refreshAttributes = scope.refreshDelay !== null ? 'refresh-delay="{{ refreshDelay }}" refresh="refreshChoices($select.search)"' : '';
+
+                    var template = `
```

Comment:
I don't understand why you repeat the `<ma-choice-field>` code instead of using the `<ma-choice-field>` directive directly.